dask_bigquery/tests/test_core.py

```diff
@@ -66,12 +71,96 @@ def dataset(df):
         )
 
 
+@pytest.fixture
+def write_dataset():
+    credentials, project_id = google.auth.default()
+    env_project_id = os.environ.get("DASK_BIGQUERY_PROJECT_ID")
+    if env_project_id:
+        project_id = env_project_id
+
+    dataset_id = uuid.uuid4().hex
+
+    yield credentials, project_id, dataset_id
+
+    with bigquery.Client() as bq_client:
+        bq_client.delete_dataset(
+            dataset=f"{project_id}.{dataset_id}",
+            delete_contents=True,
+        )
+
+
+@pytest.mark.parametrize(
+    "parquet_kwargs,load_job_kwargs",
+    [
+        (None, None),
+        (
+            {
+                "schema": pa.schema(
+                    [
+                        ("name", pa.string()),
+                        ("number", pa.uint8()),
+                        ("timestamp", pa.timestamp("ns")),
+                        ("idx", pa.uint8()),
+                    ]
+                ),
+                "write_index": False,
+            },
+            {
+                "schema": [
+                    bigquery.SchemaField("name", "STRING"),
+                    bigquery.SchemaField("number", "INTEGER"),
+                    bigquery.SchemaField("timestamp", "TIMESTAMP"),
+                    bigquery.SchemaField("idx", "INTEGER"),
+                ],
+                "autodetect": False,
+            },
+        ),
+        ({"write_index": True}, None),  # non-default
+        ({"engine": "fastparquet"}, None),  # non-default, should enforce "pyarrow"
+        (None, {"write_disposition": "WRITE_APPEND"}),  # non-default
+        (
+            None,
+            {
+                "source_format": bigquery.SourceFormat.AVRO
+            },  # non-default, should enforce PARQUET
+        ),
+    ],
+)
+def test_to_gbq(df, write_dataset, parquet_kwargs, load_job_kwargs):
+    _, project_id, dataset_id = write_dataset
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    result = to_gbq(
+        ddf,
+        project_id=project_id,
+        dataset_id=dataset_id,
+        table_id="table_to_write",
+        parquet_kwargs=parquet_kwargs,
+        load_job_kwargs=load_job_kwargs,
+    )
+    assert result.state == "DONE"
+
+
+def test_to_gbq_with_credentials(df, write_dataset):
+    credentials, project_id, dataset_id = write_dataset
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    # with explicit credentials
+    result = to_gbq(
+        ddf,
+        project_id=project_id,
+        dataset_id=dataset_id,
+        table_id="table_to_write",
+        credentials=credentials,
+    )
+    assert result.state == "DONE"
+
+
 def test_read_gbq(df, dataset, client):
     project_id, dataset_id, table_id = dataset
     ddf = read_gbq(project_id=project_id, dataset_id=dataset_id, table_id=table_id)
 
     assert list(ddf.columns) == ["name", "number", "timestamp", "idx"]
-    assert ddf.npartitions == 2
```

Comment:
Why are you removing these?

Reply:
It's not 2 anymore, it's 1. I think it's changed on the side of Big Query, the number of streams it gives back by default.

Reply:
@bnaul do you know what could be happening here? I would have expected that since we have defined a `time_partitioning` then we would be reading 2 different partitions, did something change in the default streams that you are aware of?